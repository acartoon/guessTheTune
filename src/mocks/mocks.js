import {getRandomInt, getRandomString} from "../utils";

export const settings = {
  gameTime: 5,
  errorCount: 3,
};

const ANSWERS_COUNT_GENRE = 4;
const ANSWERS_COUNT_ARTIST = 3;
const QUESTION_COUNT = 5;

const GENRES = [`rock`, `pop`, `jazz`];

const ARTISTS = [`John Snow`, `Jack Daniels`, `Jim Beam`];
// const SRC = `https://upload.wikimedia.org/wikipedia/commons/4/48/Wickethewok_-_Deepecho.ogg`;
const SRC = `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`;
const IMG_SRC = [`https://api.adorable.io/avatars/102`, `https://api.adorable.io/avatars/120`, `https://api.adorable.io/avatars/134`];

const getQuestionsGenre = () => {
  return {
    id: getRandomString(3),
    type: `genre`,
    genre: GENRES[getRandomInt(GENRES.length - 1)],
    answers: new Array(ANSWERS_COUNT_GENRE).fill(``).map(getAnswersGenre),
  };
};

const getQuestionsArtict = () => {
  return {
    id: getRandomString(3),
    type: `artist`,
    song: {
      artist: ARTISTS[getRandomInt(ARTISTS.length - 1)],
      src: SRC,
    },
    answers: new Array(ANSWERS_COUNT_ARTIST).fill(``).map(getAnswersArtict),
  };
};

const getAnswersGenre = () => {
  return {
    id: getRandomString(3),
    src: SRC,
    genre: GENRES[getRandomInt(GENRES.length - 1)],
  };
};

const getAnswersArtict = () => {
  return {
    id: getRandomString(3),
    picture: IMG_SRC[getRandomInt(IMG_SRC.length - 1)],
    artist: ARTISTS[getRandomInt(ARTISTS.length - 1)],
  };
};


export const questions = new Array(QUESTION_COUNT).fill(``).map(() => {
  return getRandomInt(1) ? getQuestionsArtict() : getQuestionsGenre();
});
