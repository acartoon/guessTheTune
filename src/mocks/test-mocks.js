export const testMocks = [
  {
    answers: [
      {id: `9mT`, src: `https://upload.wikimedia.org/wikipedia/commons/4/48/Wickethewok_-_Deepecho.ogg`, genre: `pop`},
      {id: `fRl`, src: `https://upload.wikimedia.org/wikipedia/commons/4/48/Wickethewok_-_Deepecho.ogg`, genre: `pop`},
      {id: `LIz`, src: `https://upload.wikimedia.org/wikipedia/commons/4/48/Wickethewok_-_Deepecho.ogg`, genre: `rock`},
      {id: `qJ3`, src: `https://upload.wikimedia.org/wikipedia/commons/4/48/Wickethewok_-_Deepecho.ogg`, genre: `pop`},
    ],
    genre: `jazz`,
    id: `7wO`,
    type: `genre`,
  },
  {
    answers: [
      {id: `Sub`, src: `https://upload.wikimedia.org/wikipedia/commons/4/48/Wickethewok_-_Deepecho.ogg`, genre: `pop`},
      {id: `aWl`, src: `https://upload.wikimedia.org/wikipedia/commons/4/48/Wickethewok_-_Deepecho.ogg`, genre: `pop`},
      {id: `BmX`, src: `https://upload.wikimedia.org/wikipedia/commons/4/48/Wickethewok_-_Deepecho.ogg`, genre: `pop`},
      {id: `ZGz`, src: `https://upload.wikimedia.org/wikipedia/commons/4/48/Wickethewok_-_Deepecho.ogg`, genre: `rock`},
    ],
    genre: `pop`,
    id: `30K`,
    type: `genre`,
  },
  {
    answers: [
      {id: `4G1`, picture: `https://api.adorable.io/avatars/134`, artist: `John Snow`},
      {id: `eVm`, picture: `https://api.adorable.io/avatars/102`, artist: `John Snow`},
      {id: `Gey`, picture: `https://api.adorable.io/avatars/120`, artist: `Jack Daniels`},
    ],
    id: `C9G`,
    song: {artist: `John Snow`, src: `https://upload.wikimedia.org/wikipedia/commons/4/48/Wickethewok_-_Deepecho.ogg`},
    type: `artist`
  },
  {
    answers: [
      {id: `4G1`, picture: `https://api.adorable.io/avatars/134`, artist: `John Snow`},
      {id: `eVm`, picture: `https://api.adorable.io/avatars/102`, artist: `John Snow`},
      {id: `Gey`, picture: `https://api.adorable.io/avatars/120`, artist: `Jack Daniels`},
    ],
    id: `C9G`,
    song: {artist: `John Lenon`, src: `https://upload.wikimedia.org/wikipedia/commons/4/48/Wickethewok_-_Deepecho.ogg`},
    type: `artist`
  },
];
