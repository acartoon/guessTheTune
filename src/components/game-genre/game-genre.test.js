import React from 'react';
import renderer from 'react-test-renderer';
import GameGenre from './game-genre.jsx';
import {testMocks} from '../../mocks/test-mocks';

it(`GameGenre correctly renders after relaunch`, () => {
  const tree = renderer
  .create(<GameGenre
    screenIndex={5}
    question={testMocks[0]}
    onAnswer={() => null}
  />)
  .toJSON();
  expect(tree).toMatchSnapshot();
});
