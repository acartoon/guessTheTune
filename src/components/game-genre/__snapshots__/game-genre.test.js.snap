// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameGenre correctly renders after relaunch 1`] = `
<section
  className="game game--genre"
>
  <header
    className="game__header"
  >
    <a
      className="game__back"
      href="#"
    >
      <span
        className="visually-hidden"
      >
        Сыграть ещё раз
      </span>
      <img
        alt="Угадай мелодию"
        className="game__logo"
        src="img/melody-logo-ginger.png"
      />
    </a>
    <svg
      className="timer"
      viewBox="0 0 780 780"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        className="timer__line"
        cx="390"
        cy="390"
        r="370"
        style={
          Object {
            "filter": "url(#blur)",
            "transform": "rotate(-90deg) scaleY(-1)",
            "transformOrigin": "center",
          }
        }
      />
    </svg>
    <div
      className="game__mistakes"
    >
      <div
        className="wrong"
      />
      <div
        className="wrong"
      />
      <div
        className="wrong"
      />
    </div>
  </header>
  <section
    className="game__screen"
  >
    <h2
      className="game__title"
    >
      Выберите 
      jazz
       треки
    </h2>
    <form
      className="game__tracks"
      onSubmit={[Function]}
    >
      <div
        className="track"
      >
        <div
          className="track"
        >
          <button
            className="track__button track__button--play"
            disabled={true}
            onClick={[Function]}
            type="button"
          />
          <div
            className="track__status"
          >
            <audio />
          </div>
        </div>
        <div
          className="game__answer"
        >
          <input
            checked={true}
            className="game__input visually-hidden"
            id="answer-9mT"
            name="answer"
            onChange={[Function]}
            type="checkbox"
            value="answer-pop true"
          />
          <label
            className="game__check"
            htmlFor="answer-9mT"
          >
            Отметить
          </label>
        </div>
      </div>
      <div
        className="track"
      >
        <div
          className="track"
        >
          <button
            className="track__button track__button--play"
            disabled={true}
            onClick={[Function]}
            type="button"
          />
          <div
            className="track__status"
          >
            <audio />
          </div>
        </div>
        <div
          className="game__answer"
        >
          <input
            checked={true}
            className="game__input visually-hidden"
            id="answer-fRl"
            name="answer"
            onChange={[Function]}
            type="checkbox"
            value="answer-pop true"
          />
          <label
            className="game__check"
            htmlFor="answer-fRl"
          >
            Отметить
          </label>
        </div>
      </div>
      <div
        className="track"
      >
        <div
          className="track"
        >
          <button
            className="track__button track__button--play"
            disabled={true}
            onClick={[Function]}
            type="button"
          />
          <div
            className="track__status"
          >
            <audio />
          </div>
        </div>
        <div
          className="game__answer"
        >
          <input
            checked={true}
            className="game__input visually-hidden"
            id="answer-LIz"
            name="answer"
            onChange={[Function]}
            type="checkbox"
            value="answer-rock true"
          />
          <label
            className="game__check"
            htmlFor="answer-LIz"
          >
            Отметить
          </label>
        </div>
      </div>
      <div
        className="track"
      >
        <div
          className="track"
        >
          <button
            className="track__button track__button--play"
            disabled={true}
            onClick={[Function]}
            type="button"
          />
          <div
            className="track__status"
          >
            <audio />
          </div>
        </div>
        <div
          className="game__answer"
        >
          <input
            checked={true}
            className="game__input visually-hidden"
            id="answer-qJ3"
            name="answer"
            onChange={[Function]}
            type="checkbox"
            value="answer-pop true"
          />
          <label
            className="game__check"
            htmlFor="answer-qJ3"
          >
            Отметить
          </label>
        </div>
      </div>
      <button
        className="game__submit button"
        type="submit"
      >
        Ответить
      </button>
    </form>
  </section>
</section>
`;
